### PR TITLE
2752_The_icons_of_force_get_wrong_path_for_game_play

### LINKED ISSUE #2752: The icon of force get wrong path for game play
## 🐞 Bug description
Create a new game then play game with any role. Notice  icon of force is wrong path so It can not show icon.

## 🔢 Steps to Reproduce
1. Login app as admin
2. Create a new game 
3. Play game with any role 
4. Notice  icon of force is wrong path (It gets duplicate 'http://localhost:8080'). So It can not show icon.

image

## ⚙️ Expected behavior
Should enable to show icon of force

## 💻 System information
 - Browser : Chrome
 - Browser Version : Version 120.0.6099.111

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
Fixes #2752 

## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

**Comment by lilitkarapetyan** [comment]
Fixes #2752

**Comment by Tristina1788** [comment]
The icon on the top channel are fixed. But the icon in the right bottom still show wrong. @lilitkarapetyan  Could you also fix it? Thanks.

image

**Comment by IanMayo** [comment]
Thanks @lilitkarapetyan  - a bit more tidying up, please: https://github.com/serge-web/serge/pull/2759#issuecomment-1889527013

**Comment by lilitkarapetyan** [comment]
> Thanks @lilitkarapetyan - a bit more tidying up, please: [#2759 (comment)](https://github.com/serge-web/serge/pull/2759#issuecomment-1889527013)

Hello @IanMayo, Okay I will work on it.

**Review by IanMayo — DISMISSED** [review]
Looks valid.

File: client/src/Components/local/molecules/admin-panel-footer/index.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const AdminPanelFooter: React.FC<Props> = ({ force, selectedRoleName, onI
       <div className={styles['role-type']}>{selectedRoleName}</div>
       <div className={styles['force-type']}>{force.name}</div>
     </div>
-    <img className={styles['role-icon']} src={resolveImagePath(force.iconURL || force.icon)} alt="" onClick={onIconClick} data-tour="second-step" style={{ backgroundColor: force.color }} />
+    <img className={styles['role-icon']} src={resolveImagePath(force.icon || force.iconURL)} alt="" onClick={onIconClick} data-tour="second-step" style={{ backgroundColor: force.color }} />
```
Comment (IanMayo) [inline client/src/Components/local/molecules/admin-panel-footer/index.tsx]: @lilitkarapetyan - could you remove the `icon` field from `ForceData` and then tidy up the deprecated instances please?

Comment (lilitkarapetyan) [reply client/src/Components/local/molecules/admin-panel-footer/index.tsx]: Hello @Ian, I will work on it.

**Comment by lilitkarapetyan** [comment]
We have issues with the playwright test. `cd e2e/playwright_test yarn test `

**Comment by IanMayo** [comment]
> We have issues with the playwright test. `cd e2e/playwright_test yarn test `

Yes, I think it reports that a dependency is missing.

image

@lilitkarapetyan - do you think our build could be missing a `yarn install` in the playwright folder?

**Comment by lilitkarapetyan** [comment]
> > We have issues with the playwright test. `cd e2e/playwright_test yarn test `
> 
> Yes, I think it reports that a dependency is missing.
> 
> image
> 
> @lilitkarapetyan - do you think our build could be missing a `yarn install` in the playwright folder?

I think yes.

**Review by IanMayo — DISMISSED** [review]
Tested and working

File: client/src/ActionsAndReducers/dbWargames/wargames_ActionCreators.ts [hunk 2]
```diff
@@ -306,12 +306,12 @@ export const updateForces = (dbName: string, newData: ForceData[]) => {
 }
 
 export const saveForce = (dbName: string, newData: ForceData) => {
-  return async (dispatch: WargameDispatch, state: any) => {
-    const oldForceData = state().wargame.data.forces.selectedForce
-    if (newData.iconURL !== oldForceData.iconURL && newData.iconURL !== forceTemplate.iconURL) {
-      const savedIconURL = await wargamesApi.saveIcon(newData.iconURL)
-      newData.iconURL = savedIconURL.path
-    }
+  return async (dispatch: WargameDispatch) => {
```
Comment (IanMayo) [inline client/src/ActionsAndReducers/dbWargames/wargames_ActionCreators.ts:309]: @lilitkarapetyan - this next section is commented out.  We need to decide what to do with it, please.

**Pull request merged**